src/Raven.Server/RavenServer.cs

```diff
@@ -1507,10 +1507,10 @@ public bool CanAccess(string database, bool requireAdmin, bool requireWrite)
                 // worried about race conditions here if we move to HTTP 2.0 at some point. At that point,
                 // we'll probably want to handle this concurrently, and the cost of adding it in this manner
                 // is pretty small for most cases anyway
-                _caseSensitiveAuthorizedDatabases = new Dictionary<string, DatabaseAccess>(_caseSensitiveAuthorizedDatabases)
-                {
-                    {database, mode}
-                };
+                _caseSensitiveAuthorizedDatabases = new Dictionary<string, DatabaseAccess>(_caseSensitiveAuthorizedDatabases);
+                if (_caseSensitiveAuthorizedDatabases.TryAdd(database,mode) == false)
```

Comment:
This is not thread safe, mind!

Reply:
You are creating the new instance (and making it global), then updating it.
Keep that in a local variable and then set it.